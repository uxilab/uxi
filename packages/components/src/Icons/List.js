import React from 'react';
import PropTypes from 'prop-types';
import SvgIcon from '../SvgIcon';
/* eslint-disable max-len */

const List = props => (
  <SvgIcon {...props}>
    <svg
      focusable="false"
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 16 14.2"
      width="24px"
      height="24px"
    >
      <g id="Layer_2" data-name="Layer 2">
        <g id="Layer_1-2" data-name="Layer 1">
          <rect y="4.2" width="1.59" height="1.59" />
          <rect x="3.52" y="4.2" width="12.48" height="1.59" />
          <rect y="8.4" width="1.59" height="1.59" />
          <rect x="3.52" y="8.4" width="12.48" height="1.59" />
          <rect y="12.61" width="1.59" height="1.59" />
          <rect x="3.52" y="12.61" width="12.48" height="1.59" />
          <rect width="1.59" height="1.59" />
          <rect x="3.52" width="12.48" height="1.59" />
        </g>
      </g>
    </svg>
  </SvgIcon>
);

List.propTypes = {
  color: PropTypes.string,
};

List.defaultProps = {
  color: '#6d6d71',
};

export default List;
