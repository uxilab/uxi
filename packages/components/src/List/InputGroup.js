import React from 'react';
import styled from 'styled-components';

const InputWrapper = styled.div`
  overflow: hidden;
  display: inline-flex;
  border-radius: 3px;
  border: #dcdcdc;
  width: ${({ fullWidth }) => (fullWidth ? '100%' : 'auto')};
  &>div:first-child {
    width: 100%;
  }
`;

/**
 * TODO: what happens when the context/parentDomBox
 * does not provide enough width to render all inline ? -df
 */

const InputGroup = ({ children, style = {}, fullWidth }) => (
  <InputWrapper fullWidth={fullWidth} style={style}>
    {
      React.Children.map(children, (child, i) => {
        'r';

        let rules = 0;
        if (i === 0) { rules = '3px 0 0 3px'; }
        if (i === React.Children.count(children) - 1) { rules = ' 0 3px 3px 0'; }

        return React.cloneElement(child, {
          style: {
            ...child.props.style,
            borderRadius: rules,
            minHeight: '34px',
            maxHeihgt: '34px',
            heihgt: '34px',
          },
          key: i,
        });
      })
    }
  </InputWrapper>
);

export default InputGroup;
