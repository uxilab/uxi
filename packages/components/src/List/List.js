import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

// let count = 0;
// const getKey = () => count++;

const ListWrapper = styled.div`
  margin: 0;
  padding: 0;
  list-style: none;
  li {
    display: ${({ type }) => (type === 'vertical' ? 'block' : 'inline-block')};
    padding-left: ${({ type }) => (type === 'vertical' ? '0' : '12px')};
    padding-top: ${({ type }) => (type === 'vertical' ? '6px' : '0')};
    padding-bottom: ${({ type }) => (type === 'vertical' ? '6px' : '0')};
    &:first-child {
      padding-left:0;
    }
  }
`;

/**
 *  This compo is just a "repeater" it has not styling option
 *  it "just maps stuff" and render in provided compo and DO NOT wraps it
 */
const List = ({ style, children, type }) => {
  if (React.Children.count(children) <= 0) {
    return null;
  }
  const wrappedChildren = React.Children.map(children, (c, i) => {
    const key = c.key ? `key-${c.key || c.name || c.label}` : `index-${i}`;
    return (
      <li key={key}>
        {c}
      </li>
    );
  });


  return (<ListWrapper style={style} type={type}>{wrappedChildren}</ListWrapper>);
};

List.propTypes = {
  children: [PropTypes.node],
};

List.defaultProps = {
  children: [],
};

export default List;
