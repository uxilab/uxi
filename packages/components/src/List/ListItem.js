import React from 'react';

const ListItem = ({ children, style }) => (
  <li style={style}>
    {children}
  </li>
);

export default ListItem;
