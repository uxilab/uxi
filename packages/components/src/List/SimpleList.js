import React from 'react';

const defaultListStyle = {
  margin: 0,
  padding: 0,
  listStyle: 'none',
};

const defaultItemListStyle = {
  margin: 0,
  padding: '5px 0',
};

const bulltItemListStyle = {
  listStyleType: 'disc',
  marginLeft: '20px',
  paddingLeft: '4px',
  paddingTop: '5px',
  paddingBottom: '5px',
  color: '#26a29a', // hardcode list bullet color for now, an good alternative is : https://stackoverflow.com/a/13150561  -df
};

export const SimpleList = ({ children, isBullet, style = {}, listItemStyle = {} }) => {
  const listStyle = Object.assign({}, defaultListStyle, style || {});
  let defaultListItemStyle = isBullet ? bulltItemListStyle : defaultItemListStyle;

  defaultListItemStyle = Object.assign({}, defaultListItemStyle, listItemStyle);

  const listItems = React.Children.map(children, (child, menuNumber) => {
    if (React.isValidElement(child)) {
      return React.cloneElement(child, {
        style: defaultListItemStyle,
        key: `menuItem-${menuNumber}`,
      });
    }
    return child;
  });


  return (
    <ul style={listStyle}>
      {listItems}
    </ul>
  );
};

export default SimpleList;
