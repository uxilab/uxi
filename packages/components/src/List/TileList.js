import React from 'react';
import styled from 'styled-components';

const CardListWrapper = styled.div`
  .uxi-tile-detail {
    border-top:0;
  }
  .uxi-tile-detail:first-child {
    border-top:1px solid #ccc;
  }
`;

const CardList = ({ children }) => (
  <CardListWrapper>
    {children}
  </CardListWrapper>
);

export default CardList;
