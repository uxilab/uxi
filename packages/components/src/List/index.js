/* eslint-disable import/no-named-as-default */
export default from './List';
export SimpleList from './SimpleList';
export ListItem from './ListItem';
export List from './List';
export SocialLinks from './SocialLinks';
export InputGroup from './InputGroup';
export Gallery from '../Img/Gallery';
export TileList from './TileList';
