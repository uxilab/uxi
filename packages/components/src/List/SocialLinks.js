import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { getAppropriateIcon } from '../Icons/getAppropriateIcon';

const SocialUL = styled.ul`
  display: flex;
  flex-direction: ${({ horizontal }) => (horizontal ? 'row' : 'column')};
`;

const SocialLI = styled.li`
  padding: ${({ horizontal }) => (horizontal ? '2px' : 0)};
  margin-left: ${({ horizontal, spacing }) => ((horizontal && spacing) ? spacing : 0)};
  margin-top: ${({ horizontal, spacing }) => ((!horizontal && spacing) ? spacing : 0)};
  &:first-child{
    margin-left:0;
    margin-top:0;
  }
  ${({ iconColor }) => (iconColor ?
    `svg { fill: ${iconColor}; }` : '')};
  &:hover svg {
    /* opacity: 0.9; */
    ${({ iconHoverColor }) => (iconHoverColor ? `fill: ${iconHoverColor};` : '')};
  }
`;


const SocialLinks = ({ socialLinks, horizontal, spacing, iconColor, iconHoverColor, style }) => (
  <SocialUL horizontal={horizontal} style={style}>
    {socialLinks.map(({ name, url }) => (
      <SocialLI
        horizontal={horizontal}
        iconColor={iconColor}
        iconHoverColor={iconHoverColor}
        spacing={spacing}
        key={name}
      >
        <a href={url}>
          {getAppropriateIcon(name)() }
        </a>
      </SocialLI>
    ))}
  </SocialUL>
);

SocialLinks.propTypes = {
  socialLinks: PropTypes.array,
  horizontal: PropTypes.bool,
  style: PropTypes.object,
  spacing: PropTypes.string,
  iconColor: PropTypes.string,
};

SocialLinks.defaultProps = {
  socialLinks: [],
  horizontal: false,
  style: {},
  spacing: '',
  iconColor: 0,
};

export default SocialLinks;
