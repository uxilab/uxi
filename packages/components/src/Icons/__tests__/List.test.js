import React from 'react';
import List from '../List';

describe('Icon : <List />', () => {
  it('match snapshot', () => {
    expect(shallow(<List />)).toMatchSnapshot();
  });
});

