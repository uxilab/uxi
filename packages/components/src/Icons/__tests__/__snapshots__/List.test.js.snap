// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon : <List /> match snapshot 1`] = `
<SvgIcon
  color="#6d6d71"
  hoverColor="#6d6d71"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  size={24}
  style={Object {}}
  viewBox="0 0 24 24"
>
  <svg
    focusable="false"
    height="24px"
    viewBox="0 0 16 14.2"
    width="24px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      data-name="Layer 2"
      id="Layer_2"
    >
      <g
        data-name="Layer 1"
        id="Layer_1-2"
      >
        <rect
          height="1.59"
          width="1.59"
          y="4.2"
        />
        <rect
          height="1.59"
          width="12.48"
          x="3.52"
          y="4.2"
        />
        <rect
          height="1.59"
          width="1.59"
          y="8.4"
        />
        <rect
          height="1.59"
          width="12.48"
          x="3.52"
          y="8.4"
        />
        <rect
          height="1.59"
          width="1.59"
          y="12.61"
        />
        <rect
          height="1.59"
          width="12.48"
          x="3.52"
          y="12.61"
        />
        <rect
          height="1.59"
          width="1.59"
        />
        <rect
          height="1.59"
          width="12.48"
          x="3.52"
        />
      </g>
    </g>
  </svg>
</SvgIcon>
`;
